server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -330,4 +338,106 @@ protected void doXContent(XContentBuilder builder, ToXContent.Params params) thr
             builder.field("numeric_detection", numericDetection.value());
         }
     }
+
+    public void validateDynamicTemplates(Mapper.TypeParser.ParserContext parserContext) {
+        for (DynamicTemplate dynamicTemplate : dynamicTemplates.value()) {
+            validateDynamicTemplate(parserContext, dynamicTemplate);
+        }
+    }
+
+    private static void validateDynamicTemplate(Mapper.TypeParser.ParserContext parserContext,
+                                                DynamicTemplate dynamicTemplate) {
+
+        if (containsSnippet(dynamicTemplate.getMapping(), "{name}")) {
```

Comment:
do we need to check `{dynamic_type}` too?

Reply:
Only `{name}`, because if the type hasn't specifically configured then line 358 ensures that all possible dynamic types are checked.